### PR TITLE
A more RESTful API

### PR DESCRIPTION
**Pull request by lionel-panhaleux**

### CODE REVIEW

**Comment by HenriJ** [comment]
Prefixing the id with the provider_id is a great idea, and as you said, it makes it possible to allow any kind of id and not only uuid !
Maybe we could restructure the document around resources and not endpoints ?

File: agency/README.md [hunk 1]
```diff
@@ -8,30 +8,35 @@ This specification contains a collection of RESTful APIs used to specify the dig
...
 
-## register_vehicle
+## Register vehicle
 
 The Vehicle Registration API is required in order to register a vehicle for use in the system. The API will require a valid `provider_id` and `api_key`.
 
-Endpoint: `/register_vehicle`\
-Method: `POST`\
+Endpoint: `/vehicle/{provider_id}/{unique_id}`\
+Method: `PUT`\
```
Comment (sebdiem) [inline agency/README.md]: Why use PUT here ?
In my understanding this is the moment when the vehicle object is created on the agency side => POST on `/vehicle/{provider_id}` and the unique_id is attributed via the HTTP Response.

Comment (sebdiem) [reply agency/README.md]: My bad `UUID v4 provided by Operator to uniquely identify a vehicle` the unique_id is provided by the operator.

Comment (sebdiem) [reply agency/README.md]: But still why PUT ? I still consider it as object creation on the agency side.

File: agency/README.md [hunk 2]
```diff
@@ -92,19 +102,24 @@ Response:
 | ----- | -------- | ----------------- | ----- |
 | `message` | Enum |  | See [Message](#message) Enum |
 
-## start_trip
+## Start trip
 
-Endpoint: `/start_trip`\
-Method: `POST`\
+Endpoint: `/trip/{provider_id}/{unique_id}`\
+Method: `PUT`\
```
Comment (sebdiem) [inline agency/README.md]: see above.

**Comment by sebdiem** [comment]
Great refactor IMHO ! Much more RESTful.

File: agency/README.md (hunk 1 above)
Comment (lionel-panhaleux) [reply agency/README.md]: In my understanding, PUT is used for creation and/or update, POST for adding content to an existing item (ie. a board, forum, etc.) cf. https://restfulapi.net/rest-put-vs-post/

Comment (sebdiem) [reply agency/README.md]: :+1: Right ! Thanks for the pointer !

**Pull request closed without merging**